### ISSUE TITLE
dif/2 query gives different answers on playground, local installation

### ISSUE THREAD

**Issue by jasonhemann** [aarroyoc/scryer-playground#43]
scryer's behavior for this simple query makes sense to me. 

```
[jhemann:temp] 05:27:15$ scryer-prolog --version
v0.9.4-201-gc39ea481-modified
[jhemann:temp] 05:34:56$ scryer-prolog
?- use_module(library(dif)),dif(X,Y).
   dif:dif(X,Y).
```

However, in the playground, I see the following. I get the same answer when the `use_module` is in the database.

![image](https://github.com/user-attachments/assets/10211a7c-c034-4450-aabb-228fda8d0334)

**Comment by jasonhemann** [comment]
See I think related discussion in https://github.com/aarroyoc/scryer-playground/pull/14#issue-2044844837

**Comment by bakaq** [comment]
Yeah, the toplevel in the playground doesn't handle residual goals yet. There is still a lot of work to be done here, but with https://github.com/mthom/scryer-prolog/pull/2527 I'm working towards having basically the same toplevel in the CLI and the Playground, if not even better. I think this in specific could already be solved actually, but #19 will need a lot more work.
